unittest/fortran/CMakeLists.txt

```diff
@@ -14,10 +14,12 @@ if(CMAKE_Fortran_COMPILER)
     message(STATUS "Skipping Tests for the LAMMPS Fortran Module: cannot identify Fortran compiler")
     return()
   endif()
-  # GNU Fortran 4.8.x on RHEL/CentOS 7.x is not sufficient to compile the Fortran module. Work around flang being detected as GNU
+  # GNU Fortran 4.8.x on RHEL/CentOS 7.x is not sufficient to compile the Fortran module.
+  # Also GNU Fortran 7.x on Ubuntu18.04LTS is not sufficient. Work around flang being detected as GNU
   get_filename_component(_tmp_fc ${CMAKE_Fortran_COMPILER} NAME)
-  if((CMAKE_Fortran_COMPILER_ID STREQUAL "GNU") AND (CMAKE_Fortran_COMPILER_VERSION VERSION_LESS 6.0) AND NOT (_tmp_fc STREQUAL "flang"))
-    message(FATAL_ERROR "Need GNU Fortran compiler version 6.x or later for unit testing")
+  if((CMAKE_Fortran_COMPILER_ID STREQUAL "GNU") AND (CMAKE_Fortran_COMPILER_VERSION VERSION_LESS 9.0) AND NOT (_tmp_fc STREQUAL "flang"))
```

Comment:
@hammondkd I had to raise the requirement for the Fortran compiler to GNU Fortran 9 or later, since I am having trouble compiling the Fortran module on Ubuntu 18.04LTS (with GNU Fortran 7) due to the external function support.
```
/home/akohlmey/compile/lammps/fortran/lammps.f90:2979:31:

       CALL C_F_POINTER(caller, f_caller)
                               1
Error: TS 29113/TS 18508: Noninteroperable array FPTR at (1) to C_F_POINTER: Expression is a noninteroperable derived type
/home/akohlmey/compile/lammps/fortran/lammps.f90:2950:31:

       CALL C_F_POINTER(caller, f_caller)
                               1
Error: TS 29113/TS 18508: Noninteroperable array FPTR at (1) to C_F_POINTER: Expression is a noninteroperable derived type
/home/akohlmey/compile/lammps/fortran/lammps.f90:2921:31:

       CALL C_F_POINTER(caller, f_caller)
                               1
```

I have not yet fully revised your last set of changes to the Fortran module, so I have no idea at the moment if there would be a workaround. I am ok with requiring at least Ubuntu 20.04 and later. This is likely going to become a requirement for LAMMPS with CMake after the next stable release. But it would be nice to extend the range of compatibility. Just mentioning it to you in case you might have a suggestion for what to try.

What is more worrisome is the behavior of GNU Fortran 12 with the MPI Interface that I have to work around with an added compiler flag (see below).